### PR TITLE
Fixes #31517 - do not use deprecated Net::LDAP:LdapError

### PR DESCRIPTION
**Pull request by ezr-ondrej**
`Net::LDAP::LdapError` has been deprecated in favor of `Net::LDAP::Error` since https://github.com/ruby-ldap/ruby-net-ldap/commit/c9d36cdf919e01996da5c61838c10c1bc59f3e81#diff-8e534ef69e7fb9fbb691ec029ba3883243f8ffec6c89f9fadfa13a73d5b737fb.
In https://github.com/ruby-ldap/ruby-net-ldap/commit/94b9d4d0443b8be37b91fcb5d6a31a9589cca41a#diff-8e534ef69e7fb9fbb691ec029ba3883243f8ffec6c89f9fadfa13a73d5b737fb it has got removed, so it will not be around once we upgrade to new version of `net-ldap`.
This should prevent UninitializedConstant Errors.

- [x] packaging - https://github.com/theforeman/foreman-packaging/pull/6149

### CODE REVIEW

**Comment by lzap** [comment]
Thanks, I am gonna take care of `ldap_fluff` version bump, then I will ask you to update dependency in the Gemfile once it's out.

**Comment by ezr-ondrej** [comment]
sure, can do gladly, with the packaging as well :+1: :)

**Comment by lzap** [comment]
ldap_fluff 0.5.0 is out, I bumped Y version due to breaking changes in its dependency. packaging PR in a minute.

**Comment by lzap** [comment]
* RPM: https://github.com/theforeman/foreman-packaging/pull/6149
* DEB: no packaging needed

File: Gemfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ gem 'audited', '>= 4.9.0', '< 5'
 gem 'will_paginate', '>= 3.1.7', '< 4'
 gem 'ancestry', '>= 3.0.7', '< 4', '!= 3.2.0'
 gem 'scoped_search', '>= 4.1.8', '< 5'
-gem 'ldap_fluff', '>= 0.4.7', '< 1.0'
+gem 'ldap_fluff', '>= 0.5.0', '< 1.0'
```
Comment (tbrisker) [inline Gemfile:20]: how about bumping net-ldap version as well while were at it?

Comment (ezr-ondrej) [reply Gemfile:20]: I wanted to, but it's not strictly relevant and lzap already pushed the packaging PR. I can do another one just for the `net-ldap` :thinking:

Comment (ezr-ondrej) [reply Gemfile:20]: Added just as follow-up: https://github.com/theforeman/foreman-packaging/pull/6150

Comment (ezr-ondrej) [reply Gemfile:20]: BTW ideally I'd drop `net-ldap` and hide it behind `ldap_fluff` completely - keep it only as `ldap_fluff` dependency

Comment (tbrisker) [reply Gemfile:20]: generally that would be OK but currently ldap_fluff requires >= 0.11 and we bumped it to >= 0.16 a few years ago due to a cve fix. if ldap_fluff is updated to the latest as well i'd be fine with it.

File: app/models/auth_sources/auth_source_ldap.rb [hunk 2]
```diff
@@ -266,7 +266,7 @@ def store_avatar(avatar)
   def validate_ldap_filter
     Net::LDAP::Filter.construct(ldap_filter)
   # rubocop:disable Lint/ShadowedException, Lint/RedundantCopDisableDirective
-  rescue Net::LDAP::Error, Net::LDAP::LdapError, Net::LDAP::FilterSyntaxInvalidError => e
+  rescue Net::LDAP::Error, Net::LDAP::FilterSyntaxInvalidError => e
```
Comment (tbrisker) [inline app/models/auth_sources/auth_source_ldap.rb]: Not crucial but `Net::LDAP::FilterSyntaxInvalidError <  Net::LDAP::Error`

**Review by tbrisker — APPROVED** [review]
Thanks @ezr-ondrej !

File: Gemfile (hunk 1 above)
Comment (evgeni) [reply Gemfile:20]: As long as we deploy 0.16 or newer (and don't ship a GEMFILE lock), I don't think the lower bound should matter here, as long as it's compatible?

Comment (ezr-ondrej) [reply Gemfile:20]: Yeah, I guess it's not as big of a deal... it's not like we will downgrade the package to older then 0.11 in our packaging

Comment (ezr-ondrej) [reply Gemfile:20]: I can do so in the followup PR tho :)

**Review by evgeni — APPROVED** [review]
packaging :ghost: says ACK

**Review by tbrisker — APPROVED** [review]
Thanks @ezr-ondrej and @lzap !

**Pull request merged**